src/mimeparser.rs

```diff
@@ -810,6 +811,12 @@ impl MimeMessage {
         self.headers.get(headerdef.get_headername())
     }
 
+    /// Returns `Chat-Group-ID` header value if it is a valid group ID.
+    pub fn get_chat_group_id(&self) -> Option<&String> {
```

Comment:
This could probably return the more idiomatic `&str`? Not that it matters much, feel free to ignore